src/main/kotlin/no/nav/familie/ef/sak/beregning/OmregningService.kt

```diff
@@ -0,0 +1,96 @@
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
+
+
+@Service
+class OmregningService(private val behandlingService: BehandlingService,
+                       private val fagsakService: FagsakService,
+                       private val vedtakService: VedtakService,
+                       private val iverksettService: IverksettService,
+                       private val taskRepository: TaskRepository,
+                       private val iverksettClient: IverksettClient,
+                       private val ytelseService: TilkjentYtelseService,
+                       private val grunnlagsdataService: GrunnlagsdataService,
+                       private val featureToggleService: FeatureToggleService,
+                       private val vurderingService: VurderingService,
+                       private val beregnYtelseSteg: BeregnYtelseSteg,
+                       private val iverksettingDtoMapper: IverksettingDtoMapper) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    fun utførGOmregning(forrigeBehandlingId: UUID) {
+
+        feilHvisIkke(featureToggleService.isEnabled("familie.ef.sak.omberegning")) {
+            "Feature toggle for omberegning er disabled"
+        }
+
+        val fagsak = fagsakService.fagsakMedOppdatertPersonIdent(forrigeBehandlingId)
+        val sisteBehandling = behandlingService.finnSisteBehandling(fagsak.id)
+        feilHvisIkke(sisteBehandling?.erAvsluttet() == true) {
+            "Kan ikke omberegne fagsak med åpen behandling, id=${sisteBehandling!!.id}"
+        }
+
+        val behandling = behandlingService.opprettBehandling(behandlingType = BehandlingType.REVURDERING,
+                                                             fagsakId = fagsak.id,
+                                                             behandlingsårsak = BehandlingÅrsak.G_OMREGNING)
+        logger.info("G-omregner fagsak=${fagsak.id} behandling=${behandling.id} ")
+
+
+        val forrigeTilkjentYtelse = ytelseService.hentForBehandling(forrigeBehandlingId)
+
+        val vedtak = vedtakService.hentVedtak(forrigeBehandlingId)
+
+
+        iverksettService.startBehandling(behandling, fagsak)
+
+        grunnlagsdataService.opprettGrunnlagsdata(behandling.id)
+        vurderingService.opprettVilkårForOmregning(behandling)
+
+        val vedtaksperioder = vedtak.mapInnvilgelseOvergangsstønad().perioder
+        val indeksjusterInntekt =
+                BeregningUtils.indeksjusterInntekt(forrigeTilkjentYtelse.grunnbeløpsdato, vedtak.inntekter?.inntekter ?: listOf())
+        val saksbehandling = behandlingService.hentSaksbehandling(behandling.id)
+
+        beregnYtelseSteg.utførSteg(saksbehandling, InnvilgelseOvergangsstønad(periodeBegrunnelse = null,
```

Comment:
Liten tanke: Går det å gjøre noen form av sjekk i`beregnYtelseSteg` som sjekker att første dato for endret andel er 1 mai eller noe slik?
Kan ju gjøres utenom denne PR :)

Reply:
Yes, tar i egen PR :)